### PR TITLE
chore: remove usage of "export *"

### PR DESCRIPTION
**Pull request by layershifter**
## Previous Behavior

`export *` is used in a lot of files.

## New Behavior

**No API changes**

`export *` is replaced with exact exports. This will improve performance of current compiler in Griffel and will unblock the future one, see https://github.com/microsoft/griffel/pull/621.

This PR also enables the ESLint rule for `**/src/**/*.{ts,tsx,js}` files, so we won't regress.

Note 1: I used autofix via `--fix` on ESLint, so some other things _were_ fixed (unused ESLint supressions, etc.).
Note 2: This change affects some infra packages (like `scripts`), but IMO they don't deserve a separate config. Let's follow best practices everywhere.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

File: packages/eslint-plugin/src/configs/base.js [hunk 1]
```diff
@@ -19,10 +19,9 @@ module.exports = {
   },
```
Comment (fabricteam) [inline packages/eslint-plugin/src/configs/base.js:19]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv8 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33448&commitId=826bb409967f86a001dbedeb43edc329945190ce&env=prod&clientType=FLUENTUI)


  <details><summary><b>Callout</b> 6 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Callout.Beak 25.chromium.png|2186| Changed|
|Callout.Bottom Left Edge RTL.chromium.png|2178| Changed|
|Callout.Right center.chromium.png|2396| Changed|
|Callout.Top center.chromium.png|2178| Changed|
|Callout.Top right edge.chromium.png|1594| Changed|
|Callout.Left bottom edge.chromium.png|3022| Changed|

</details>


  <details><summary><b>react-charting-AreaChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Custom Accessibility.chromium.png|11| Changed|

</details>


  <details><summary><b>react-charting-LineChart</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-LineChart.Events.chromium.png|2| Changed|
|react-charting-LineChart.Gaps.chromium.png|1| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

Comment (fabricteam) [inline packages/eslint-plugin/src/configs/base.js:19]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the FluentUIV0 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33448&commitId=826bb409967f86a001dbedeb43edc329945190ce&env=prod&clientType=FLUENTUI)


  <details><summary><b>Card</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card.CardExampleSimple.chromium.png|224| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v0"/>

**Comment by fabricteam** [comment]
## 🕵 FluentUIV0 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v0"/>

File: packages/eslint-plugin/src/configs/base.js (hunk 1 above)
Comment (fabricteam) [inline packages/eslint-plugin/src/configs/base.js:19]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33448&commitId=7f62a0148d259bb7c557bf6524bdf7a3298cf713&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.badgeMask.chromium.png|9| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-icon-provider/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,9 @@
 import './version';
 
-export * from './IconProvider';
-export * from './IconProvider.types';
+export {
+  // eslint-disable-next-line @fluentui/ban-context-export
```
Comment (layershifter) [inline packages/react-icon-provider/src/index.ts:4]: It's already exported ¯\_(ツ)_/¯

File: apps/vr-tests-react-components/src/utilities/index.ts [hunk 3]
```diff
@@ -1,3 +1,9 @@
-export * from './TestWrapperDecorator';
-export * from './getStoryVariant';
-export * from './withStoryWrightSteps';
+export {
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/utilities/index.ts:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentui-web-components-v3 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33448&commitId=9d98c53a9f063b2c1a95b9bbc4bafa266c0c49b4&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar.Image.chromium.png|248| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Review by ling1726 — APPROVED** [review]

**Review by mainframev — APPROVED** [review]

**Review by sopranopillow — APPROVED** [review]

**Review by khmakoto — APPROVED** [review]

**Pull request merged**